auth/tests/desktop/rpcs/create_auth_uri_test.cc

```diff
@@ -41,6 +50,28 @@ TEST(CreateAuthUriTest, TestCreateAuthUriRequest) {
       request.options().post_fields);
 }
 
+#if FIREBASE_PLATFORM_DESKTOP
+TEST(CreateAuthUriTest, TestCreateAuthUriRequestWithHeartbeatPayload) {
+  std::unique_ptr<App> app(testing::CreateApp());
+  ::firebase::heartbeat::DateProviderImpl date_provider;
+  Logger logger(nullptr);
+  HeartbeatStorageDesktop storage(app->name(), logger);
+  HeartbeatController controller(app->name(), logger, date_provider);
+  // For the sake of testing, clear any pre-existing stored heartbeats.
+  LoggedHeartbeats empty_heartbeats_struct;
+  storage.Write(empty_heartbeats_struct);
+  // Then log a single heartbeat for today's date.
+  controller.LogHeartbeat();
+  CreateAuthUriRequest request(*app, "APIKEY", "email");
+
+  // The payload will be encoded and the date will be today's date
+  // But it should at least be non-empty.
+  EXPECT_NE("", request.options().header[app_common::kApiClientHeader]);
+  EXPECT_EQ("com.google.firebase.testing",
+            request.options().header[app_common::kXFirebaseGmpIdHeader]);
+}
+#endif  // FIREBASE_PLATFORM_DESKTOP
+
```

Comment:
Ideally we should test this for EVERY `AuthRequest` that will deliver the heartbeat but I am expecting this to fail if this type of test runs more than once because
1. The heartbeat storage persists per app_id per machine. If this test run more than once, you will get an empty payload
2. We have cooldown to get payload.  If this test run in sequence, ex. due to retry, this will fail.

I have a feeling that this new test may introduce more trouble than what it can help.

I recommend to add a new Buganizer task to create tests to verify if Auth request actually carries heartbeat payload properly.